### PR TITLE
(PRODEV-9292) Security Scan Skip

### PR DESCRIPTION
**Pull request by dylanTesouro**
Motivation
---
We want to skip security scans for images that are just used locally.

Modifications
---
- Add arg to skip security scan of images
- Bump GHA versions that are using deprecated node versions

Results
---
Skipping protection where trusted

### CODE REVIEW

File: common-actions/github-app-token/action.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ runs:
   steps:
     - name: 'GitHub App Token'
       id: generate_token
-      uses: tibdex/github-app-token@b62528385c34dbc9f38e5f4225ac829252d1ea92
+      uses: tibdex/github-app-token@3eb77c7243b85c65e84acfa93fdbac02fb6bd532
```
Comment (Tesouro-Chris) [inline common-actions/github-app-token/action.yml:19]: Can we just use the tag `v2.1.0` instead please? 
https://github.com/tibdex/github-app-token/releases/tag/v2.1.0

Comment (dylanTesouro) [reply common-actions/github-app-token/action.yml:19]: When I had asked @Sam-tesouro about that, he mentioned that for security purposes its best to use the specific commit hash. OR I just made that conversation up in my head.

Comment (Tesouro-Chris) [reply common-actions/github-app-token/action.yml:19]: Uh, if true, we have about 300 other places to update to commit hashes then

Comment (dylanTesouro) [reply common-actions/github-app-token/action.yml:19]: Sorry, I mean for this specifically, since it has to do with security.

Comment (Tesouro-Chris) [reply common-actions/github-app-token/action.yml:19]: Right, sorry, but I mean
`aws-actions/amazon-ecr-login@v2`
`aws-actions/configure-aws-credentials@v4`
`docker/login-action@v3`

There's a lot of these that would deal with security elements imho. Maybe not 300, but like.... a good amount

Comment (Sam-tesouro) [reply common-actions/github-app-token/action.yml:19]: this tibdex repo was maintained by a third-party who I don't have as much trust in Chris

Comment (Sam-tesouro) [reply common-actions/github-app-token/action.yml:19]: albeit it was referenced by docker in one of their docs

Comment (Tesouro-Chris) [reply common-actions/github-app-token/action.yml:19]: Gotcha, yeah, this is one of the main council dudes I think. https://jupyter.org/ I'm good with keeping a hash though. Shall resolve.

File: functional-tests-ecr/action.yml [hunk 2]
```diff
@@ -76,6 +76,15 @@ runs:
...
         
+    - name: Restore Docker Image Cache
+      uses: actions/cache@v3
+      with:
+        path: /tmp/docker-cache
+        key: ${{ runner.os }}-docker-${{ hashFiles('Dockerfile', 'docker-compose.ci.yml') }}-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-docker-${{ hashFiles('Dockerfile', 'docker-compose.ci.yml') }}-
+          ${{ runner.os }}-docker-
+
```
Comment (dylanTesouro) [inline functional-tests-ecr/action.yml:87]: @Tesouro-Chris was this what you had in mind? I wasn't sure about a specific list of components. 

https://tesourohq.slack.com/archives/C0736MEH4BY/p1723649977122689?thread_ts=1723648409.235379&cid=C0736MEH4BY

Comment (dylanTesouro) [reply functional-tests-ecr/action.yml:87]: Answer, no, this wont work
`Warning: No files were found with the provided path: /tmp/docker-cache. No artifacts will be uploaded.`
🤔

Comment (dylanTesouro) [reply functional-tests-ecr/action.yml:87]: We shaved off the bulk of the time with reusing images, there is probably more to do, but im going to close this as good enough for now, CD becomes the choking point anyway.

**Review by Sam-tesouro — APPROVED** [review]
blessed_doge

**Pull request merged**